File: S5/INFO503 - Modélisation client-serveur et programmation Web avancée/TP/TP4/Classes/Utilisateur.php
<?php
class Utilisateur{
  protected $id;
  protected $login;
  protected $mdp;

  function __construct(int $id, string $login, string $mdp){
    $this->id = $id;
    $this->login = $login;
    $this->mdp = $mdp;

  }

  function __toString():string{
    return "ID : ".$this->getId()."\nLogin : ".$this->getLogin()."\nMot de passe : ".$this->getMdp();
  }

  function getId():int{
      return $this->id;
  }

  function getLogin():string{
    return $this->login;
  }

  function getMdp():string{
    return $this->mdp;
  }
}

File: S5/INFO503 - Modélisation client-serveur et programmation Web avancée/TP/projet_sondages/src/Utilisateur.java
import org.json.*;

public class Utilisateur{
  protected String login;
  protected String password;
  protected static int id = 0;

  public Utilisateur(String login, String password){
    this.login = login;
    this.password = password;
    ++Utilisateur.id;
  }

  public int getId(){
    return Utilisateur.id;
  }
  public String getLogin(){
    return this.login;
  }
  public String getPassword(){
    return this.password;
  }

  public String toString(){
    return login+" , "+password+" , "+Utilisateur.id;
  }

  public JSONObject toJSON(){
    JSONObject j = new JSONObject(this);
    return j;
  }
}
